src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/AceEditorWidget.java

```diff
@@ -267,29 +268,12 @@ public void onAttachOrDetach(AttachEvent event)
          }
       });
       
-      events_.addHandler(
-            BeginPasteEvent.TYPE,
-            new BeginPasteEvent.Handler()
-            {
-               
-               @Override
-               public void onBeginPaste(BeginPasteEvent event)
-               {
-                  maybeUnmapCtrlV();
-               }
-            });
-      
-      events_.addHandler(
-            EndPasteEvent.TYPE,
-            new EndPasteEvent.Handler()
-            {
-               
-               @Override
-               public void onEndPaste(EndPasteEvent event)
-               {
-                  maybeRemapCtrlV();
-               }
-            });
+      if (!hasPasteHandlers_)
```

Comment:
Note that only one copy of these events need to be registered as they modify the 'global' Emacs / Vim command bindings, which are shared by all Ace instances.